software/contrib/preset_manager.py

```diff
@@ -0,0 +1,191 @@
+from time import sleep
+from europi import *
+from europi_script import EuroPiScript
+
+
+class PresetManager(EuroPiScript):
+    def __init__(self):
+        super().__init__()
+        state = self.load_state_json()
+
+        self.blank_preset = ["blank preset", [0,0,0,0,0,0]]
+
+        self.presets = state.get("presets", [["preset one", [0,0,0,0,0,0]], ["preset two", [0,0,0,0,0,0]], ["preset three", [0,0,0,0,0,0]]]) # [[preset name, [preset values]], [preset name 2, [preset values 2]]
+        self.loaded_preset = state.get("loaded preset", 0)
+        self.selected_preset = k1.read_position(7) - 1
+        self.selected_letter = 0
+        self.screen = 1
+        
+        self.old_k2 = 0
+        
+        self.resolution = 128
+        
+        self.box_width = int(OLED_WIDTH/3)
+        self.box_height = int(OLED_HEIGHT/2.8)
+        
+        self.list_line_height = CHAR_HEIGHT + 4
+        
+        self.output_multiplier = MAX_OUTPUT_VOLTAGE / self.resolution
+        
+        b1.handler(self.back_button)
+        b2.handler(self.enter_button)
+        
+        self.characters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ' ', ' ']
+        self.max_characters = 16
+
+
+    def save_state(self):
+        """Save the current state variables as JSON."""
+        # Don't save if it has been less than 5 seconds since last save.
+        if self.last_saved() < 5000:
+            return
+        state = {
+            "presets": self.presets,
+            "loaded preset": self.loaded_preset
+        }
+        self.save_state_json(state)
+        
+        
+    def back_button(self):
+        self.save_state()
+        
+        self.screen -= 1
+        
+        if self.screen == -3: #Confirm delete
+            self.presets.remove(self.presets[self.hovered_preset])
+            self.screen = 0
+            self.update_ui()
```

Comment:
It's better to keep your handler functions as light weight as possible. Consider using a flag like `self.should_update=True` instead (here and anywhere else that triggers the need to update UI) and inside your `update_ui()` method, bail out if should update is False:

```
def update_ui(self):
    if !self.should_update:
        return
    ...
```